### PR TITLE
[RHDX-410] Akamai Cache Clear block

### PR DESCRIPTION
**Pull request by ghost**
### JIRA Issue Links
* https://projects.engineering.redhat.com/browse/RHDX-410
* https://projects.engineering.redhat.com/browse/RHDX-411

### Verification Process
0. Login as an Admin (also test for Content Admin)
1. Go to a page using the rhdp2 theme and page.html.twig template: https://developer-preview-3540.ext.us-west.dc.preprod.paas.redhat.com/
2. See the "Refresh Page in Akamai" button next to the login button
3. Clicking the button should result in a success message
4. Go to a page using the rhdp2 theme and one of each of the:
* Katacoda Course page: https://developer-preview-3540.ext.us-west.dc.preprod.paas.redhat.com/courses/spring-boot/
* Katacoda Lesson page: https://developer-preview-3540.ext.us-west.dc.preprod.paas.redhat.com/courses/spring-boot/monitoring-applications/
* Kiosk Mode page: https://developer-preview-3540.ext.us-west.dc.preprod.paas.redhat.com/devnation/master-course/
5. Repeat steps 2 & 3 above on these pages.
6. Logout
7. Repeat steps 1 &4
8. You should NOT see the "Refresh Page in Akamai" button

#### Page
Screen Shot 2020-04-07 at 7 21 31 AM

#### Katacodas & Kiosk
Screen Shot 2020-04-07 at 7 21 42 AM

### CODE REVIEW

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

File: _docker/drupal/drupal-filesystem/web/config/sync/user.role.content_admin.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ permissions:
   - 'access site in maintenance mode'
   - 'access site reports'
   - 'add assembly entities'
+  - 'administer akamai'
```
Comment (staceymosier) [inline _docker/drupal/drupal-filesystem/web/config/sync/user.role.content_admin.yml:19]: Is this a necessary permission for content_admin?

File: _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.theme [hunk 2]
```diff
@@ -21,6 +23,39 @@ function rhdp2_preprocess_page(&$variables) {
   $variables['main_menu'] = _rhdp2_build_menu('rhd-navigation', ['pf-c-nav__tertiary-list']);
   $variables['topics_menu'] = _rhdp2_build_menu('topics-menu', ['pf-c-nav__tertiary-list']);
   $variables['mobile_menu'] = _rhdp2_build_menu('rhd-navigation', ['pf-c-nav__tertiary-list'], 'menu__rhd_navigation__mobile');
+  $variables['akamai_cache_clear_button'] = _rhdp2_build_block('akamai_cache_clear_block');
+}
+
+ /**
```
Comment (staceymosier) [inline _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.theme:29]: There are several PHPCs warnings for this block
- Line indented incorrectly; expected 0 spaces, found 1
- Parameter tags must be grouped together in a doc comment
- Type hint "array" missing for $config

Can you please clear the phpcs warnings here and the ones introduced by the ics changes?

File: _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.theme [hunk 3]
```diff
@@ -21,6 +23,39 @@ function rhdp2_preprocess_page(&$variables) {
...
+  *
+  * @return array
+  *   Returns a render array for the block or an empty array.
+  */
+function _rhdp2_build_block($bid = '', $config = []) {
+
+  // Get the block.
+  $block_service = \Drupal::service('plugin.manager.block');
+  $block = $block_service->createInstance($bid, $config);
+  if ($block instanceof Broken) {
```
Comment (staceymosier) [inline _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.theme:47]: Neat!

**Review by staceymosier — CHANGES_REQUESTED** [review]
Please see comments.

File: _docker/drupal/drupal-filesystem/web/config/sync/user.role.content_admin.yml (hunk 1 above)
Comment (rogeralloymagnetic) [reply _docker/drupal/drupal-filesystem/web/config/sync/user.role.content_admin.yml:19]: Probably not, I may have gone a little "permissions checkbox happy" with that one.

**Comment by staceymosier** [comment]
@rogeralloymagnetic   Can you also update the block roles that appears on in the RHDP theme for Product pages?

/admin/structure/block/manage/akamaicacheclear

File: _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.theme (hunk 2 above)
Comment (rogeralloymagnetic) [reply _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.theme]: Resolved in commit d72870a

File: _docker/drupal/drupal-filesystem/web/config/sync/user.role.content_admin.yml (hunk 1 above)
Comment (rogeralloymagnetic) [reply _docker/drupal/drupal-filesystem/web/config/sync/user.role.content_admin.yml]: Resolved in commit 22eb3c1

**Review by staceymosier — APPROVED** [review]
Thank you!

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by robcoffman** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by rogeralloymagnetic** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
I verified that the RHEL download e2e test failure is a false negative. I ran through the workflow manually, and I was able to successfully download the boot iso file. Let me try to retest this PR now.

**Comment by jordanpagewhite** [comment]
retest this please

**Comment by jordanpagewhite** [comment]
I just realized that this will need to rebuild either way once we resolve the `rhdp2.theme` file merge conflict

**Comment by jordanpagewhite** [comment]
I just resolved the merge conflict and pushed up the changes. Hopefully it builds successfully and the tests pass and we can move straight into reviews tomorrow morning.

**Comment by rogeralloymagnetic** [comment]
retest this please

**Review by jordanpagewhite — APPROVED** [review]
I just verified this for the following routes:

* https://developer-preview-3540.ext.us-west.dc.preprod.paas.redhat.com/
* https://developer-preview-3540.ext.us-west.dc.preprod.paas.redhat.com/products/rhel/overview
* https://developer-preview-3540.ext.us-west.dc.preprod.paas.redhat.com/products/openjdk/overview
* https://developer-preview-3540.ext.us-west.dc.preprod.paas.redhat.com/courses/spring-boot/
* https://developer-preview-3540.ext.us-west.dc.preprod.paas.redhat.com/courses/spring-boot/monitoring-applications/
* https://developer-preview-3540.ext.us-west.dc.preprod.paas.redhat.com/devnation/master-course/

and the button was present, on all routes, when I was authenticated to my jowhite Drupal user account, and was not present when I was unauthenticated.

I verified that I could successfully click the button and trigger a cache clear, for the current route, in Akamai successfully on the homepage.

File: _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.theme [hunk 4]
```diff
@@ -21,6 +23,40 @@ function rhdp2_preprocess_page(&$variables) {
...
+    return [];
+  }
+
+  // Check access.
+  $user_access = $block->access(\Drupal::currentUser());
+  $access_denied = ($user_access === FALSE) ? FALSE : TRUE;
+  $access_denied = ($user_access instanceof AccessResult && $user_access->isForbidden()) ? FALSE : TRUE;
```
[the remark this answers is not in the document]
Comment (rogeralloymagnetic) [reply _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.theme]: @jordanpagewhite I'm not sure why this comment or code block is showing up now. Maybe you're looking at an old commit, or this message is off in the timeline. The current version of the code already has a fix implemented for this: 
https://github.com/redhat-developer/developers.redhat.com/pull/3540/files?file-filters%5B%5D=.js&file-filters%5B%5D=.theme&file-filters%5B%5D=.twig#diff-304ef1e9f8b0e28569a944480f0512da

**Pull request merged**